File: go/src/274.h-指数.go
/*
 * @lc app=leetcode.cn id=274 lang=golang
 *
 * [274] H 指数
 */

// @lc code=start
func hIndex(citations []int) int {
	n := len(citations)
	sort.Ints(citations)
	for i := 0; i < n; i++ {
		// 当引用次数比剩余论文数量还多的时候，继续增大引用次数，只会减小论文数量，进而导致h变小
		if n-i <= citations[i] {
			return n - i
		}
	}

	return 0
}

// @lc code=end

File: rust/src/274.h-指数.rs
/*
 * @lc app=leetcode.cn id=274 lang=rust
 *
 * [274] H 指数
 */

// @lc code=start
impl Solution {
    pub fn h_index(mut citations: Vec<i32>) -> i32 {
        citations.sort();
        let n = citations.len() as i32;
        for i in 0..n {
            if citations[i as usize] >= n - i {
                return n - i;
            }
        }

        0
    }
}
// @lc code=end
